programs/pluto/kernel_xfrm_interface.c

```diff
@@ -369,7 +370,7 @@ static ip_cidr get_xfrmi_ipaddr_from_conn(struct connection *c, struct logger *l
 			ldbg(logger,
 				 "get_xfrmi_ipaddr_from_conn() taking IP from sourceip param for xfrmi IF [%s] id [%d]",
 				 c->xfrmi->name, c->xfrmi->if_id);
-			return cidr_from_address_prefix_len(*sip, 32);
+			return cidr_from_address_prefix_len(*sip, (address_info(*sip) == &ipv4_info ? 32 : 128));
```

Comment:
Since these are the only calls to `cidr_from_address_prefix_len()` how about changing the interface so it works better. `ip_cidr cidr_from_address(ip_address)` say.  It could use `struct ip_info` contains the field `unsigned mask_cnt; /* 32 or 128 */`.  Want me to make that change?

Reply:
sure, thanks.

Reply:
pushed, it will appear shortly